models/issues/milestone.go

```diff
@@ -344,6 +424,7 @@ func (milestones MilestoneList) getMilestoneIDs() []int64 {
 type GetMilestonesOption struct {
 	db.ListOptions
 	RepoID   int64
+	Repo     *repo_model.Repository
```

Comment:
I don't think we need this.

Reply:
It is needed in [toLabel](https://github.com/go-gitea/gitea/blob/eaae336808cb80cf5628f540d0f877bc171458ac/modules/convert/issue.go#L192), [ToAPIMilestone](https://github.com/go-gitea/gitea/blob/eaae336808cb80cf5628f540d0f877bc171458ac/modules/convert/issue.go#L222), [NewMilestoneLabels](https://github.com/go-gitea/gitea/blob/eaae336808cb80cf5628f540d0f877bc171458ac/models/issues/milestone_label.go#L73).
Should these values be made into separate properties of GetMilestonesOption?

Reply:
[Here](https://github.com/go-gitea/gitea/blob/eaae336808cb80cf5628f540d0f877bc171458ac/models/issues/label.go#L620), and in [issue](https://github.com/go-gitea/gitea/blob/eaae336808cb80cf5628f540d0f877bc171458ac/models/issues/issue.go#L111) for example repo is also added

Reply:
I mean `Repo` should be assigned out of the function.

Reply:
Can you please specify what it means to be assigned out of the function?

Reply:
Remove this new field of this struct and assign repo after invoking this function.